File: doc-examples/example-java/src/main/java/example/aws/S3ClientBuilderCustomizer.java
package example.aws;

import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;
import software.amazon.awssdk.services.s3.S3ClientBuilder;

import java.time.Duration;
import java.time.temporal.ChronoUnit;

//tag::class[]
@Singleton
public class S3ClientBuilderCustomizer implements BeanCreatedEventListener<S3ClientBuilder> {

    @Override
    public S3ClientBuilder onCreated(@NonNull BeanCreatedEvent<S3ClientBuilder> event) {
        return event.getBean()
            .overrideConfiguration(c -> c.apiCallTimeout(Duration.of(60, ChronoUnit.SECONDS)));
    }
}
//end::class[]

File: doc-examples/example-python/src/main/python/example/aws/S3ClientBuilderCustomizer.py
from jakarta.inject import Singleton
from java.time import Duration
from java.time.temporal import ChronoUnit
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
from software.amazon.awssdk.services.s3 import S3ClientBuilder


# tag::class[]
@Singleton
class S3ClientBuilderCustomizer(BeanCreatedEventListener[S3ClientBuilder]):

    def onCreated(self, event: BeanCreatedEvent[S3ClientBuilder]) -> S3ClientBuilder:
        return event.getBean() \
            .overrideConfiguration(lambda c: c.apiCallTimeout(Duration.of(60, ChronoUnit.SECONDS)))
# end::class[]
